src/sandstorm/run-bundle.c++

```diff
@@ -1270,6 +1313,23 @@ private:
       }
     }
 
+    // Now process the PORT setting, since the actual value in config.ports
+    // depends on if HTTPS_PORT was provided at any point in reading the
+    // config file.
+    //
+    // Outer KJ_IF_MAYBE so we only run this code if the config file contained
+    // a PORT= declaration.
+    KJ_IF_MAYBE(portValue, maybePortValue) {
+        KJ_IF_MAYBE(ports, getPorts(config.httpsPort, *portValue)) {
+            // Memory leak? Who owns this pointer? I guess it has to stay
+            // alive as long as the program does, since we don't typically
+            // delete the Config object.
+            config.ports = kj::mv(*ports);
+        } else {
+          KJ_FAIL_REQUIRE("invalid config value PORT", *portValue);
```

Comment:
On the more general subject of gotos, I think that the pattern you're talking about applies primarily to C where you're basically implementing the "throw exception" code path manually. In particular, you have to clean up all the resources that you've allocated during the function. In C++ we get that automatically via destructors. I would say that using "goto" for error handling in C++ is probably wrong most of the time.

I am not totally against gotos, though. One place where I find they make sense is for implementing "retry", especially after EINTR. (Of course, in KJ you usually use KJ_SYSCALL which handles EINTR automatically, but sometimes I need to make a syscall manually.)